CMakeLists.txt

```diff
@@ -469,6 +464,14 @@ target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURR
 if (BUILD_TESTING)
     enable_testing()
 
+    # FIPS mode is optional for Openssl and must be deliberately configured
+    if ("$ENV{S2N_LIBCRYPTO}" MATCHES "openssl" AND "$ENV{S2N_LIBCRYPTO}" MATCHES "fips")
```

Comment:
I do worry that this is a bit too "magical". 

Matching on `S2N_LIBCRYPTO` is very CI-specific logic, so I'd prefer that configuration to actually live in our buildspec.

E.g. one CI run with `openssl-1.0.2-fips` with `-DS2N_FIPS_TEST=ON` enabled, and another with `-DS2N_FIPS_TEST=OFF` enabled. Then we could drop the "test for all fips modes" logic from CMake.